### PR TITLE
Feature/core 975 nameresolver.get address returns undefined for undefined address

### PR DESCRIPTION
**Pull request by OmairLiaquatAli**

### CODE REVIEW

File: src/name-resolver.spec.ts [hunk 1]
```diff
@@ -78,4 +80,15 @@ describe('NameResolver class', function test() {
...
     expect(address).to.eq(emptyAddressValue);
   });
+
+  it('should throw an error when resolving ENS name undefined', async () => {
+    const nameResolver = await TestUtils.getNameResolver(web3);
+    const name = '';
+    const address = nameResolver.getAddress(name);
+    const contents = nameResolver.getContent(name);
+    const addressOrContents = nameResolver.getAddressOrContent(name, 'address');
+    await (expect(address).to.be.rejectedWith('ens name is undefined. Provide valid ens name. example ens name: xyz.evan'));
```
Comment (Philser) [inline src/name-resolver.spec.ts]:
```suggestion
    await expect(address).to.be.rejectedWith('ens name is undefined. Provide valid ens name. example ens name: xyz.evan');
```

File: src/name-resolver.spec.ts [hunk 2]
```diff
@@ -78,4 +80,15 @@ describe('NameResolver class', function test() {
...
+    await (expect(contents).to.be.rejectedWith('ens name is undefined. Provide valid ens name. example ens name: xyz.evan'));
```
Comment (Philser) [inline src/name-resolver.spec.ts]:
```suggestion
    await expect(contents).to.be.rejectedWith('ens name is undefined. Provide valid ens name. example ens name: xyz.evan');
```

File: src/name-resolver.spec.ts [hunk 3]
```diff
@@ -78,4 +80,15 @@ describe('NameResolver class', function test() {
...
+    await (expect(addressOrContents).to.be.rejectedWith('ens name is undefined. Provide valid ens name. example ens name: xyz.evan'));
```
Comment (Philser) [inline src/name-resolver.spec.ts]:
```suggestion
    await expect(addressOrContents).to.be.rejectedWith('ens name is undefined. Provide valid ens name. example ens name: xyz.evan');
```

File: src/name-resolver.ts [hunk 4]
```diff
@@ -72,6 +72,9 @@ export class NameResolver extends Logger {
    */
   public async getAddressOrContent(name: string, type: string): Promise<string> {
     this.log(`looking up ENS name "${name}"`, 'debug');
+    if (!name) {
+      throw new Error('ens name is undefined. Provide valid ens name. example ens name: xyz.evan');
```
Comment (Philser) [inline src/name-resolver.ts]:
```suggestion
      throw new Error('ENS name is undefined. Provide valid ENS name. Example ENS name: xyz.evan');
```

Comment (Philser) [reply src/name-resolver.ts]: Also, if you choose to apply this suggestion, keep in mind to adapt the expected error messages in the tests

File: src/name-resolver.ts [hunk 5]
```diff
@@ -96,6 +99,9 @@ export class NameResolver extends Logger {
    * @return     Promise, that resolves to {string} address
    */
   public async getAddress(name: string) {
+    if (!name) {
+      throw new Error('ens name is undefined. Provide valid ens name. example ens name: xyz.evan');
```
Comment (Philser) [inline src/name-resolver.ts]:
```suggestion
      throw new Error('ENS name is undefined. Provide valid ENS name. Example ENS name: xyz.evan');
```

File: src/name-resolver.ts [hunk 6]
```diff
@@ -107,6 +113,9 @@ export class NameResolver extends Logger {
    * @return     Promise, that resolves to {string} address
    */
   public async getContent(name: string) {
+    if (!name) {
+      throw new Error('ens name is undefined. Provide valid ens name. example ens name: xyz.evan');
```
Comment (Philser) [inline src/name-resolver.ts]:
```suggestion
      throw new Error('ENS name is undefined. Provide valid ENS name. Example ENS name: xyz.evan');
```

File: src/name-resolver.spec.ts (hunk 3 above)
Comment (Philser) [inline src/name-resolver.spec.ts]:
```suggestion
    await expect(addressOrContents).to.be.rejectedWith('ens name is undefined. Provide valid ens name. example ens name: xyz.evan');
```

File: src/name-resolver.ts [hunk 7]
```diff
@@ -96,6 +99,9 @@ export class NameResolver extends Logger {
    * @return     Promise, that resolves to {string} address
    */
   public async getAddress(name: string) {
+    if (!name) {
```
Comment (wulfraem) [inline src/name-resolver.ts]: Do we need this check here? This function is using `getAddressOrContent`, where we have another check.

File: src/name-resolver.ts [hunk 8]
```diff
@@ -107,6 +113,9 @@ export class NameResolver extends Logger {
    * @return     Promise, that resolves to {string} address
    */
   public async getContent(name: string) {
+    if (!name) {
```
Comment (wulfraem) [inline src/name-resolver.ts]: Do we need this check here? This function is using `getAddressOrContent`, where we have another check.

File: src/name-resolver.ts (hunk 7 above)
Comment (OmairLiaquatAli) [reply src/name-resolver.ts]: no need for it. Ran the test without the checks here. Passes.

File: src/name-resolver.spec.ts [hunk 9]
```diff
@@ -78,4 +80,15 @@ describe('NameResolver class', function test() {
     address = await nameResolver.getAddress(testAddress2);
     expect(address).to.eq(emptyAddressValue);
   });
+
+  it.only('should throw an error when resolving ENS name undefined', async () => {
```
Comment (wulfraem) [inline src/name-resolver.spec.ts]:
```suggestion
  it('should throw an error when resolving ENS name undefined', async () => {
```

**Pull request merged**